soroban-env-host/src/host/crypto.rs

```diff
@@ -77,13 +84,129 @@ impl Host {
         })
     }
 
+    #[cfg(feature = "next")]
+    pub(crate) fn secp256r1_verify_signature(
+        &self,
+        verifying_key: &p256::ecdsa::VerifyingKey,
+        msg_hash: &Hash,
+        sig: &Signature<p256::NistP256>,
+    ) -> Result<(), HostError> {
+        let _span = tracy_span!("p256 verify");
+        self.charge_budget(ContractCostType::VerifyEcdsaSecp256r1Sig, None)?;
+        verifying_key
+            .verify_prehash(msg_hash.as_slice(), sig)
+            .map_err(|_| {
+                self.err(
+                    ScErrorType::Crypto,
+                    ScErrorCode::InvalidInput,
+                    "failed scep256r1 verification",
+                    &[],
+                )
+            })
+    }
+
+    #[cfg(feature = "next")]
+    pub(crate) fn secp256r1_decode_sec1_uncompressed_pubkey(
+        &self,
+        bytes: &[u8],
+    ) -> Result<p256::ecdsa::VerifyingKey, HostError> {
+        use sec1::point::Tag;
+        self.charge_budget(ContractCostType::Sec1DecodePointUncompressed, None)?;
+        // check and make sure the key was encoded in uncompressed format
+        let tag = bytes
+            .first()
+            .cloned()
```

Comment:
nit: `.copied()`? we just have a single byte here and it would look less suspicious (my first intuition when I see `clone` is that we have a bigger data structure)

Reply:
fixed